internal/driver/driver.go

```diff
@@ -277,6 +281,70 @@ func reportOptions(p *profile.Profile, vars variables) (*report.Options, error)
 	return ropt, nil
 }
 
+// inferNumericLabelUnits returns a map of numeric label keys to the units
+// associated with those keys.
+// Unit for a given key is the first encountered unit for that key. If multiple
+// units are encountered for values paired with a particular key, then the first
+// unit encountered is used and an error message is displayed.
+// if units are encountered for a particular key, the unit is then inferred
+// based on the key.
+func inferNumericLabelUnits(p *profile.Profile, ui plugin.UI) map[string]string {
```

Comment:
"infer" is a bit misleading since we only infer the unit if it's not specified explicitly and actually use the unit from the profile when it's there?